src/facade/disk_backed_queue.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2025, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <io/file.h>
+
+#include <functional>
+#include <memory>
+#include <string_view>
+#include <system_error>
+
+#include "io/io.h"
+
+namespace facade {
+
+class DiskBackedQueue {
+ public:
+  explicit DiskBackedQueue(uint32_t conn_id);
+  ~DiskBackedQueue();
+
+  std::error_code Init();
+
+  // Check if we can offload bytes to backing file.
+  bool HasEnoughBackingSpaceFor(size_t bytes) const;
+
+  std::error_code Write(io::Bytes bytes);
```

Comment:
@romange we would need a flag in dfly_conn because a fiber blocking disk write can't be called in the cb passed to RegisterRecv. The other option would be to go async here but IMO not worth it